consent-ws/src/main/java/org/broadinstitute/consent/http/service/ontology/IndexerUtils.java

```diff
@@ -1,35 +1,107 @@
-package org.broadinstitute.consent.http.service.ontologyIndexer;
+package org.broadinstitute.consent.http.service.ontology;
 
 import com.google.common.collect.Lists;
 import com.twitter.util.CountDownLatch;
+import org.apache.commons.io.IOUtils;
 import org.apache.http.HttpEntity;
 import org.apache.http.entity.ContentType;
 import org.apache.http.nio.entity.NStringEntity;
+import org.broadinstitute.consent.http.models.ontology.StreamRec;
 import org.broadinstitute.consent.http.models.ontology.Term;
 import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
 import org.elasticsearch.client.ResponseListener;
 import org.elasticsearch.client.RestClient;
+import org.semanticweb.owlapi.apibinding.OWLManager;
 import org.semanticweb.owlapi.model.*;
 import org.semanticweb.owlapi.reasoner.OWLReasoner;
+import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
+import org.semanticweb.owlapi.reasoner.structural.StructuralReasonerFactory;
+import org.semanticweb.owlapi.search.EntitySearcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.InternalServerErrorException;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
- * A suite of functions for handling index functions.
+ * A suite of functions for handling ontology indexing functions.
  */
 public class IndexerUtils {
 
-    private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(IndexerUtils.class);
+    private static final String FIELD_DEFINITION_PROPERTY = "IAO_0000115";
+    private static final String FIELD_HAS_EXACT_SYNONYM_PROPERTY = "hasExactSynonym";
+    private static final String FIELD_LABEL_PROPERTY = "label";
+    private static final String FIELD_DEPRECATED_PROPERTY = "deprecated";
+    private static final Logger logger = LoggerFactory.getLogger(IndexerUtils.class);
+    private static final ArrayList<String> IRI_FILTERS = new ArrayList<>(Arrays.asList("DOID", "DUOS", "DUO"));
```

Comment:
Note: This approach is less than ideal. Without filtering, we get a range of undesired terms from DUO that only add confusion to the index. Even if we move this to a configuration, that still requires a code update. I think a better solution would be to accept a list of IRI filter terms from the API ... looking for reviewer feedback on this question and if we should add a story for updating the API to accept an optional filter list.

Reply:
I agree feels not-ideal ... but practically, we don't change the "whitelist" of ontologies very often, so I don't mind this as-is.